src/main/java/com/github/cameltooling/dap/internal/model/scopes/CamelExchangeScope.java

```diff
@@ -66,7 +66,11 @@ public SetVariableResponse setVariableIfInScope(SetVariableArguments args, Manag
 		if (getVariablesReference() == args.getVariablesReference()) {
 			throw new UnsupportedOperationException("Not supported");
 		}
-		return exchangeVariable.setVariableIfInScope(args, backlogDebugger);
+		if (exchangeVariable != null) {
+			return exchangeVariable.setVariableIfInScope(args, backlogDebugger);
+		} else {
+			return null;
+		}
```

Comment:
If the default case is returning null is much better to have it at the end of the function instead of inside an else. The else provides no value here.

In fact I think is much better to use a short circuit approach like

```
-Stopping conditions (throw, return nulls)
-Environment
-Process
-Return result
```

Which would be like

```
		if (getVariablesReference() == args.getVariablesReference()) {
			throw new UnsupportedOperationException("Not supported");
		}
		if (exchangeVariable == null)  return null; 
		

	    return exchangeVariable.setVariableIfInScope(args, backlogDebugger);
		
```

Reply:
modified to have the return null before. using an else if as the order of condition check is important. So seems like it is a way to emphasize it.

Reply:
But you will never reach that part of the code, as all of those are short circuit conditions so entering any if means leaving the normal check flow. But I get what you mean and I think it's much better now